### PR TITLE
correct value argument name for Tensor.index_fill_ docs

### LINKED ISSUE #51250: Wrong argument name of index_fill_
## 📚 Documentation

https://pytorch.org/docs/stable/tensors.html#torch.Tensor.index_fill_

The third argument should be called `value` instead of `val`, like all other methods in `torch.Tensor`.

Using `val` throws an error and the error message indicate this should be `value` in the background implementation:
`* (name dim, Tensor index, Tensor value)
      didn't match because some of the keywords were incorrect: val`

cc @jlin27 @mruberry

### PR DESCRIPTION
**Pull request by dagitses**
Summary:
The name of "val" is inconsistent with the rest of the API and also
inconsistent with the underlying C++ implementation.

Test Plan:
Used the following command to demonstrate incorrect docs before and
correct docs after:
  python -c 'import torch; print(torch.Tensor.index_fill_.__doc__)'

Reviewers: alband

Subscribers: mruberry loveritsu929

Tasks: #51250

Tags:

Fixes #51250

### CODE REVIEW

**Review by albanD — APPROVED** [review]
LGTM!
